tests/loops/test_loops.py

```diff
@@ -693,26 +704,26 @@ def test_fit_loop_reset(tmpdir):
 
     fit_loop.load_state_dict(mid_epoch_ckpt["loops"]["fit_loop"])
 
-    def mid_epoch_reset_assertions():
+    def mid_epoch_reset_assertions(has_reset):
         assert fit_loop.restarting
         assert fit_loop.epoch_progress.total.ready == 1
         assert fit_loop.epoch_progress.total.completed == 0  # the checkpoint was saved mid epoch
-        assert fit_loop.epoch_progress.current.ready == 0
+        assert fit_loop.epoch_progress.current.ready == (not has_reset)
         assert fit_loop.epoch_progress.current.completed == 0
 
         assert epoch_loop.restarting
         assert epoch_loop.batch_progress.total.ready == 2
         assert epoch_loop.batch_progress.total.completed == 1  # the checkpoint was saved on train_batch_end
-        assert epoch_loop.batch_progress.current.ready == 2
-        assert epoch_loop.batch_progress.current.completed == 2
+        assert epoch_loop.batch_progress.current.ready == 2 if has_reset else 1
+        assert epoch_loop.batch_progress.current.completed == 2 if has_reset else 1
 
     # resetting from a mid-epoch checkpoint should not change progress counters
```

Comment:
the comment here says it. the mid epoch checkpoint here is a complete one (no failure) and therefore the counters are never different. resetting should not change any values.

Reply:
Yes, it would now as the restarting logic is moved in reset

Reply:
Before it would be done automatically while reloading. @carmocca argued it should be moved in reset function

Reply:
the way the test changes here just makes it not meaningful anymore. we want our tests to reflect the expected behavior and to avoid regressions. 

I'm changing it

Reply:
@awaelchli there is value in having the checks before reset to make sure we load exactly what was saved

Reply:
that's already covered in other tests